internal/test/e2e/tinkerbellHardwareQueue.go

```diff
@@ -0,0 +1,58 @@
+package e2e
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/eks-anywhere/internal/pkg/api"
+)
+
+// Default timeout for Tink tests to poll for hardware.
+const hwPollingTimeout = 120 * time.Minute
+
+type hardwareQueue struct {
+	hws []*api.Hardware
+	mu  sync.Mutex
+}
+
+func (hwQu *hardwareQueue) reserveHardware(count int) ([]*api.Hardware, error) {
+	now := time.Now()
+	after := now.Add(hwPollingTimeout)
+	for {
+		if now.After(after) {
+			return nil, fmt.Errorf("hardware polling request timed out")
+		}
+		hwQu.mu.Lock()
+		if count <= len(hwQu.hws) {
+			hardwareReserved := hwQu.dequeueHw(count)
+			hwQu.mu.Unlock()
```

Comment:
nit: do a defer unlock after the lock on line 27.

Reply:
defer unlock wouldn't work here as this is a polling function that has to periodically poll to check for hardware availability. We do not want to wait until the timeout to unlock as there could be tests that need lesser number hardware waiting for the lock.?